fuzz/v3name_fuzzer.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2012-2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include <openssl/e_os2.h>
+#include <openssl/x509.h>
+#include <openssl/x509v3.h>
+#include "internal/nelem.h"
+#include "fuzzer.h"
+
+int FuzzerInitialize(int *argc, char ***argv)
+{
+    return 1;
+}
+
+int FuzzerTestOneInput(const uint8_t* data, size_t size){     
+    GENERAL_NAME *namesa = OPENSSL_malloc(size);
+    GENERAL_NAME *namesb = OPENSSL_malloc(size);
+
+    if (namesa == NULL || namesb == NULL)
+        return 0;
```

Comment:
There is no need to allocate or check this. The d2i_GENERAL_NAME() below will allocate it, and so the ones above get leaked.

Reply:
done.